src/NanopassSharp/Patterns/BinaryPattern.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+
+namespace NanopassSharp.Patterns;
+
+/// <summary>
+/// A pattern which applies a binary operation to two other patterns.
+/// </summary>
+public sealed class BinaryPattern : ITransformationPattern
+{
+    /// <summary>
+    /// The left pattern.
+    /// </summary>
+    public ITransformationPattern Left { get; }
+
+    /// <summary>
+    /// The right pattern.
+    /// </summary>
+    public ITransformationPattern Right { get; }
+
+    /// <summary>
+    /// The binary operation.
+    /// </summary>
+    public Func<bool, bool, bool> Operation { get; }
+
+    public bool IsRecursive =>
+        Left.IsRecursive || Right.IsRecursive;
+
+
+
```

Comment:
I don't know if this is copypaste or intentional, but you have a couple of these blank line blocks around.